src/NzbDrone.Core/Download/Clients/DownloadStation/Proxies/DiskStationProxyBase.cs

```diff
@@ -105,115 +179,109 @@ private void AuthenticateClient(DownloadStationSettings settings)
 
-            var downloadStationResponse = Json.Deserialize<DiskStationResponse<DiskStationAuthResponse>>(response.Content);
-
-            var authResponse = Json.Deserialize<DiskStationResponse<DiskStationAuthResponse>>(response.Content);
-
-            _authenticated = authResponse.Success;
-
-            if (!_authenticated)
+            foreach (var arg in arguments)
             {
-                throw new DownloadClientAuthenticationException(downloadStationResponse.Error.GetMessage(DiskStationApi.Auth));
+                requestBuilder.AddQueryParam(arg.Key, arg.Value);
             }
+
+            var authResponse = ProcessRequest<DiskStationAuthResponse>(DiskStationApi.Auth, requestBuilder.Build(), settings, "login");
+
+            return authResponse.Data.SId;
         }
 
-        private HttpRequest BuildRequest(DownloadStationSettings settings, DiskStationApi api, Dictionary<string, object> arguments, HttpMethod method)
+        private HttpRequestBuilder BuildRequest(DownloadStationSettings settings, DiskStationApiInfo apiInfo, HttpMethod method)
         {
-            if (!Resources.ContainsKey(api))
-            {
-                GetApiVersion(settings, api);
-            }
+            var requestBuilder = new HttpRequestBuilder(settings.UseSsl, settings.Host, settings.Port).Resource($"webapi/{apiInfo.Path}");
 
-            var requestBuilder = new HttpRequestBuilder(settings.UseSsl, settings.Host, settings.Port).Resource($"webapi/{Resources[api]}");
             requestBuilder.Method = method;
             requestBuilder.LogResponseContent = true;
             requestBuilder.SuppressHttpError = true;
             requestBuilder.AllowAutoRedirect = false;
 
-            if (requestBuilder.Method == HttpMethod.POST)
-            {
-                if (api == DiskStationApi.DownloadStationTask && arguments.ContainsKey("file"))
-                {
-                    requestBuilder.Headers.ContentType = "multipart/form-data";
-
-                    foreach (var arg in arguments)
-                    {
-                        if (arg.Key == "file")
-                        {
-                            Dictionary<string, object> file = (Dictionary<string, object>)arg.Value;
-                            requestBuilder.AddFormUpload(arg.Key, file["name"].ToString(), (byte[])file["data"]);
-                        }
-                        else
-                        {
-                            requestBuilder.AddFormParameter(arg.Key, arg.Value);
-                        }
-                    }
-                }
-                else
-                {
-                    requestBuilder.Headers.ContentType = "application/json";
-                }
-            }
-            else
-            {
-                foreach (var arg in arguments)
-                {
-                    requestBuilder.AddQueryParam(arg.Key, arg.Value);
-                }
-            }
+            return requestBuilder;
```

Comment:
As mentioned, you already know the api here via apiInfo, so afaik you can add the &api= queryparam.